src/libraries/Common/src/System/Reflection/TypeNameParser.cs

```diff
@@ -0,0 +1,632 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Text;
+
+namespace System.Reflection
+{
+    //
+    // Parser for type names passed to GetType() apis.
+    //
+    internal ref partial struct TypeNameParser
+    {
+        private TypeNameParser(ReadOnlySpan<char> name)
+        {
+            _input = name;
+            _errorIndex = _index = 0;
+        }
+
+        //
+        // Parses a type name. The type name may be optionally postpended with a "," followed by a legal assembly name.
+        //
+        private Type? Parse()
+        {
+            TypeName? typeName = ParseNonQualifiedTypeName();
+            if (typeName is null)
```

Comment:
I believe it is matter of personal performance, unless it is one of those cases where the difference matters. `== null` / `!= null` is more popular than `is`/`is not` equivalents currently. `is not null` is the least popular - it is not surprising since it is more characters than the shorter equivalent.

Reply:
I have updated everything to is / is not null.